Sources/MembraneRTC/Media/SoundDetection.swift

```diff
@@ -0,0 +1,179 @@
+import AVFoundation
+
+// Class for sound detection using the device's microphone
+public class SoundDetection: ObservableObject {
+
+  // Audio engine for sound detection
+  private var engine: AVAudioEngine!
+  private var inputNode: AVAudioInputNode!
+  private var mixerNode: AVAudioMixerNode!
+  private var onSoundDetectedListener: ((_ detectionResult: Bool) -> Void)?
+  private var onVolumeChangedListener: ((_ volume: Int) -> Void)?
+
+  // Property to indicate the recording state
+  @Published public var isRecording: Bool = false
+
+  /// Sets a listener to receive sound volume change events.
+  ///
+  /// - Parameter listener: The listener to be notified when sound volume changes.
+  public func setOnVolumeChangedListener(listener: ((_ volume: Int) -> Void)?) {
```

Comment:
`(_ volume: Int) -> Void)` can we alias it to a type? so that you don't have to write `(_ volume: Int) -> Void)` every time but for example `VolumeChangedListener`
Is it aliased on Android?